### PR TITLE
fix empty topics in a non-breaking way

### PR DESCRIPTION
**Pull request by yihuang**
related: https://github.com/tharsis/ethermint/pull/840

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: rpc/ethereum/backend/utils.go [hunk 1]
```diff
@@ -155,5 +155,11 @@ func TxLogsFromEvents(events []abci.Event) ([]*ethtypes.Log, error) {
 			logs = append(logs, &log)
 		}
 	}
-	return evmtypes.LogsToEthereum(logs), nil
+	ethLogs := evmtypes.LogsToEthereum(logs)
+	for _, log := range ethLogs {
+		if log.Topics == nil {
+			log.Topics = make([]common.Hash, 0)
```
Comment (calvinaco) [inline rpc/ethereum/backend/utils.go:161]: It is safer to do

```
			ethLogs[i].Topics = make([]common.Hash, 0)
```

**Review by calvinaco — COMMENTED** [review]
Left a comment on the range loop.

File: rpc/ethereum/backend/utils.go [hunk 2]
```diff
@@ -155,5 +155,11 @@ func TxLogsFromEvents(events []abci.Event) ([]*ethtypes.Log, error) {
 			logs = append(logs, &log)
 		}
 	}
-	return evmtypes.LogsToEthereum(logs), nil
+	ethLogs := evmtypes.LogsToEthereum(logs)
```
Comment (thomas-nguy) [inline rpc/ethereum/backend/utils.go:158]: nitpicking but maybe you can also improve performance of `LogsToEthereum` 

by initializing 

```
ethLogs := make([]common.Hash, len(logs))
...

ethLogs[i] = logs[i].ToEthereum()

File: rpc/ethereum/backend/utils.go (hunk 1 above)
Comment (yihuang) [reply rpc/ethereum/backend/utils.go]: done

**Review by calvinaco — APPROVED** [review]
LGTM

**Comment by thomas-nguy** [comment]
Can you add an entry to the changelog?

**Pull request merged**